### DISCUSSION TITLE
Are struct entries invalid?

### DISCUSSION THREAD

**Discussion by lolgear** [libgit2/libgit2#5977] [Q&A]
`indexer.c` and `xpatience.c` define `struct entry`.

```c
/// indexer.c
struct entry {
  git_oid oid;
  uint32_t crc;
  uint32_t offset;
  uint64_t offset_long;
};

/// xpatience.c
struct hashmap {
  int nr, alloc;
  struct entry {
    unsigned long hash;
    /*
     * 0 = unused entry, 1 = first line, 2 = second, etc.
     * line2 is NON_UNIQUE if the line is not unique
     * in either the first or the second file.
     */
    unsigned long line1, line2;
    /*
     * "next" & "previous" are used for the longest common
     * sequence;
     * initially, "next" reflects only the order in file1.
     */
    struct entry *next, *previous;

    /*
     * If 1, this entry can serve as an anchor. See
     * Documentation/diff-options.txt for more information.
     */
    unsigned anchor : 1;
  } *entries, *first, *last;
  /* were common records found? */
  unsigned long has_matches;
  mmfile_t *file1, *file2;
  xdfenv_t *env;
  xpparam_t const *xpp;
};
```

I guess these entries should be renamed to `git_indexer_entry` and `xpatience_hashmap_entry` respectively.

```
libgit2Package/SwiftPackage/Sources/Libgit2Origin/src/indexer.c:31:8: Type 'struct entry' has incompatible definitions in different translation units

/// Previous definition
/// usr/include/search.h
typedef	struct entry {
	char	*key;
	void	*data;
} ENTRY;

```

**Comment by carlosmn** [comment]
What system and compiler is saying this? The `search.h` file is only included in `src/util.c` that I can find so they wouldn't conflict. They may well have the same name in different translation units, but they don't interact.

**Reply by lolgear** [reply]
It's a -fmodules flag.
If I pass -fno-modules ( current solution ), this flag will be disabled and everything works fine.
I guess it happens because clang doesn't define modules for C and tries to compile library against Darwin module.
Not sure if I'm correct here, I wish clang (and spm) has better flags discoverability.

**Comment by lolgear** [comment]
Passing additional `-fno-modules` flag solves current problem.

**Reply by lolgear** [reply]
But this flag is "unsafe" and it doesn't solve a problem for anyone who wants to use libgit2 as SPM.

**Reply by lolgear** [reply]
... for anyone who wants to compile libgit2 with -fmodules flag
